worlds/sc2/rules.py

```diff
@@ -219,6 +219,46 @@ def great_train_robbery_train_stopper(self, state: CollectionState) -> bool:
                 or state.has_any({item_names.VULTURE, item_names.LIBERATOR}, self.player)
             )
         )
+    
+    def zerg_great_train_robbery_train_stopper(self, state: CollectionState) -> bool:
+        """
+        Ability to deal with trains (moving target with a lot of HP)
+        :param state:
+        :return:
+        """
+        return (
+            self.morph_impaler_or_lurker(state)
+            or state.has_all({item_names.MUTALISK_SEVERING_GLAIVE, item_names.MUTALISK}, self.player)
+            or state.has(item_names.ABERRATION, self.player)
+            or state.has(item_names.MALIGNANT_CREEP, self.player)
+            or self.advanced_tactics
+            and (
```

Comment:
(Minor, optional) recommend moving `and (` to be at the end of the previous line, just to keep the indentation level always 'or'. It makes the associativity a little easy to parse visually.

If you change this, recommend changing it for terran and protoss functions as well for consistency. I think this pattern (indentation level is either 'or' or 'and', not both) should be fairly consistent in the file nowadays.

Reply:
Use:
```python
or (self.advanced_tactics
            and (
...
```
(wrap the advanced tactics rule into brackets)

Reply:
Fixed formatting to what Ziktofel recommended